lib/util/builder.ts

```diff
@@ -0,0 +1,48 @@
+import * as BalenaSdk from '../../typings/balena-sdk';
+import { InjectedDependenciesParam, InjectedOptionsParam } from '../balena';
+
+interface BuilderBuildFromUrlSuccessResponse {
+	started: true;
+	releaseId: number;
+}
+
+type BuilderBuildFromUrlResponse =
+	| {
+			started: false;
+			error: string;
+			message: string;
+	  }
+	| BuilderBuildFromUrlSuccessResponse;
+
+export class BuilderHelper {
+	constructor(
+		private deps: InjectedDependenciesParam,
+		private opts: InjectedOptionsParam,
+	) {}
+
+	public buildFromUrl(
+		owner: string,
+		appName: string,
+		urlDeployOptions: BalenaSdk.BuilderUrlDeployOptions,
+	) {
+		return this.deps.request
+			.send<BuilderBuildFromUrlResponse>({
+				method: 'POST',
+				url: `/v3/buildFromUrl?headless=true&owner=${owner}&app=${appName}`,
+				baseUrl: this.opts.builderUrl,
+				body: Object.assign(
+					{
+						shouldFlatten: true,
+					},
+					urlDeployOptions,
+				),
+			})
+			.then(resp => {
+				const data = resp.body;
+				if ('error' in data) {
```

Comment:
NIT: in most places we would do `data[error]`

Reply:
That was mostly for TS to be able to reason about the union types, but I will change it to check for truthiness :+1:

Reply:
After adding the tests, it seems this only has to be inside a `.catch` since balena-request will reject all promises with non 2xx statusCode.